synadm/api.py

```diff
@@ -409,6 +409,22 @@ def user_list(self, _from, _limit, _guests, _deactivated,
             "user_id": _user_id
         })
 
+    def user_list_paginate(self, _limit, _guests, _deactivated,
+                           _name, _user_id, _from="0"):
+        """Yields API responses for all of the pagination.
+
+        Args:
+            _from (string): Opaque string
+        """
+        while _from is not None:
+            response = self.user_list(_from, _limit, _guests, _deactivated,
+                                      _name, _user_id)
+            yield response
+            if "next_token" in response.keys():
+                _from = response["next_token"]
```

Comment:
because of how clever this code is, the `_from` variable is modified. this can have undesirable effects for callers passing their valuable variables:
```
>>> def asdf(_asdf):
...     _asdf += ["i am not supposed to be in your var"]
...     return _asdf
...
>>> a = ['hi']
>>> a
['hi']
>>> asdf(a)
['hi', 'i am not supposed to be in your var']
>>> a
['hi', 'i am not supposed to be in your var']
```

this applies to lists (which are not expected) and maybe other things too.